src/error.rs

```diff
@@ -0,0 +1,43 @@
+use platform_impl;
+
+pub enum ExternalError {
+    NotSupported(NotSupportedError),
+    Os(OsError),
+}
+
+/// TODO: MANUALLY IMPLEMENT TRAITS
+#[derive(Debug)]
+pub struct NotSupportedError {
+    _marker: (),
```

Comment:
What's the point of this marker?

Reply:
So that `NotSupportedError` is an opaque type. That has two purposes:
1. Users can't create instances of it
2. We can freely add fields to it in the future without making breaking changes.